group_utils/README.md

```diff
@@ -0,0 +1,171 @@
+Perform arithmetic operations on the states of Number members. 
+This functionality is the same as in the openHAB [group item definition](https://www.openhab.org/docs/configuration/items.html#derive-group-state-from-member-items).
+The following functions only use Items of type: Number, Dimmer or Rollershutter.
+Other Items and NULL or UNDEF states are ignored.
+
+### MembersSum
+Get the __SUM__ of direct members' number states.
+```javascript
+var sum = MembersSum(group)
+```
+
+### MembersAvg
+Get the __AVERAGE__ of direct members' number states.
+```javascript
+var avg = MembersAvg(group)
+```
+
+### MembersMin
+Get the __MINIMUM__ of direct members' number states.
+```javascript
+var min = MembersMin(group)
+```
+
+### MembersMax
+Get the __MAXIMUM__ of direct members' number states.
+```javascript
+var max = MembersMax(group)
+```
+
+### allMembersSum
+Get the __SUM__ of all (also child) members' number states.
+```javascript
+var sum = allMembersSum(group)
+```
+
+### allMembersAvg
+Get the __AVERAGE__ of all (also child) members' number states.
+```javascript
+var avg = allMembersAvg(group)
+```
+
+### allMembersMin
+Get the __MINIMUM__ of all (also child) members' number states.
+```javascript
+var min = allMembersMin(group)
+```
+
+### allMembersMax
+Get the __MAXIMUM__ of all (also child) members' number states.
+```javascript
+var max = allMembersMax(group)
+```
+
+### __Count operations__
+
+### countMembers
+Count the direct members matching a given comparison function. 
+Comparison is done through a user-passed function.
+The function returns how many members match the given comparison function.
+```javascript
+var counter = countMembers(group, compareFunc)
+// with the following example scheme:
+var counter = countMembers(group, function (i) { return i.getState() /* comparison operators */ })
+
+// examples
+var lightsOFF = countMembers('lights', function (i) { return i.getState() == OFF }) // the number of lights off
+var larg24 = GroupUtils.count('temperatures', return i.getState() > 24 ) // the number of temperatures higher than 24
+var smalEq24 = GroupUtils.count('temperatures', function (i) { return i.getState() <= 24 }) // the number of temperatures lower or equal than/to 24
+```
+Argument | Purpose | Required
+-|-|-
+`group` | The name of the group. | yes
+`compareFunc` | The comparison function, pass and use `i` to work with the item.  | yes
+
+### countAllMembers
+Count direct and child group members matching a given comparison function.
+The group items are excluded.
+The function returns how many members match the given comparison function.
+```javascript
+var counter = countAllMembers(group, compareFunc)
+// with the following example scheme:
+var counter = countAllMembers(group, function (i) { return i.getState() /* comparison operators */ })
+```
+The arguments are exactly the same as for [countMembers](#countmembers).
+The examples work in the same way as for [countMembers](#countmembers).
+
+### __Testing__
```

Comment:
Let's remove all mention of testing until we actually implement some tests. I don't want to leave the impression that the library has automated tests when it's still a TBD. I'll merge this without the tests.